pkg/workflows/wasm/host/wasip1.go

```diff
@@ -41,9 +44,34 @@ func newWasiLinker(engine *wasmtime.Engine) (*wasmtime.Linker, error) {
 		return nil, err
 	}
 
+	err = linker.FuncWrap(
+		"wasi_snapshot_preview1",
+		"random_get",
+		randomGet,
+	)
+	if err != nil {
+		return nil, err
+	}
+
 	return linker, nil
 }
 
+func randomGet(caller *wasmtime.Caller, buf, bufLen int32) int32 {
+	// Get slice of memory for the buffer
+	bufPtr := &buf
+	bufFromPtr := (*[4]byte)(unsafe.Pointer(bufPtr))[:]
+
+	// Fix the random source with a hardcoded seed
+	randSource := rand.New(rand.NewSource(42)) //nolint:gosec
```

Comment:
@MStreet3 There should be some functions in this package that will allow you to interact with the buffers/pointers in a less slow level way (slightly less anyway).

Check out:
- safeMem to copy a buffer from memory
- copyBuffer to take a buffer and copy it into memory

In this case you probably want to create a new buffer with bufLen and then copy it into the WASM memory using copyBuffer